server/src/main/java/org/elasticsearch/index/mapper/ScriptRuntimeField.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+public abstract class ScriptRuntimeField implements RuntimeField {
```

Comment:
I am a bit on the fence on the naming, mostly because the object field is also going to hold a script but will not extend this class. Maybe this could be named LeafRuntimeField or something along those lines?

Reply:
++ to LeafRuntimeField, the main use of this base class is to simplify returning a single-valued collection from `asMappedFieldTypes()` and 'Leaf' reinforces that.